base/experimental.jl

```diff
@@ -457,4 +457,18 @@ without adding them to the global method table.
 """
 :@MethodTable
 
+"""
+    Base.Experimental.entrypoint(f, argtypes::Tuple)
```

Comment:
What is the difference between `entrypoint` and `precompile`?
Could we just use `precompile`, or add a keyword to it?

The docstring here should include the "where code not provably reachable from entry points is removed." bit

Reply:
The problem is we want to mark only the minimal needed entry points, and not compile anything speculatively, so existing precompile statements should be ignored. I guess it could be a keyword argument, but only in the sense that any function can be turned into any other function by adding a keyword argument :shrug: